server/apps/publish/formatters/aap_ipnews_formatter.py

```diff
@@ -122,3 +174,110 @@ def format(self, article, subscriber):
+        if 'NEWSLIST' in article['slugline'].upper():
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name, 'sportmedia', 'sportseds')
+            return
+        if category['qcode'] == 'T' and article['source'] == 'AAP' and article['urgency'] > 3:
+            odbc_item['selector_codes'] = self.SELECTOR_CODES['sportd'][subscriber_name]
+            return
+        if category['qcode'] == 'T' and article['source'] == 'AAP' and article['urgency'] <= 3:
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name, 'sportd', 'csportd')
+            return
+        if category['qcode'] == 'S' and article['source'] == 'AAP':
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name, 'sporti', 'csporti', 'csportd')
+            return
+        if category['qcode'] == 'S' and article['source'] != 'AAP' and article['urgency'] <= 3:
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name, 'sporti', 'csporti')
+            return
+        if category['qcode'] == 'S' and article['source'] != 'AAP' and article['urgency'] > 3:
+            odbc_item['selector_codes'] = self.SELECTOR_CODES['sporti'][subscriber_name]
+            return
+
+    def _set_fdx_selector_codes(self, article, subscriber_name, odbc_item, category):
+        if category['qcode'] == 'F' and article['priority'] in ['N'] and article['urgency'] > 3:
+            odbc_item['selector_codes'] = self.SELECTOR_CODES['sfinanced'][subscriber_name]
+            return
+        if category['qcode'] == 'F' and article['source'] == 'AAP' and article['priority'] in ['N', 'M']:
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name,
+                                                                    'financed', 'cfinanced', 'sfinanced')
+            return
+        if category['qcode'] == 'F' and article['source'] != 'AAP':
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name,
+                                                                    'financei', 'cfinancei', 'sfinancei')
+            return
+        if category['qcode'] == 'F' and article['source'] == 'AAP' and article.get('locator', '') not in ['N', 'M']:
+            odbc_item['selector_codes'] = self._join_selector_codes(subscriber_name,
+                                                                    'financei', 'cfinancei', 'sfinancei')
+            return
+
+    def _join_selector_codes(self, subscriber_name, *args):
+        codes = []
+        for arg in args:
+            codes.extend(self.SELECTOR_CODES[arg][subscriber_name.lower()].split())
+        return ' '.join(list(set(codes)))
+
+    def _is_in_subject(self, article, qcode):
+        def compare(code1, code2):
+            return code1[:len(code2)] == code2
+
+        return any([compare(s['qcode'], qcode) for s in article.get('subject', [])])
```

Comment:
No real need for the (temporary) list here, a generator expression as a parameter is already enough.